src/app/core/services/logging.service.ts

```diff
@@ -16,19 +17,30 @@ export class LoggingService {
     }
   }
 
+  saveToLocalStorage(...params: any[]) {
+    const oldestSessionLogRegex = /(AppConfig,.*\n[\S\s]*)(?=AppConfig,.*)/g;
```

Comment:
I'm not sure if regex is the way to go, but even so, I think it is only necessary to perform this replacement in the constructor/during initialization.

Since we want maybe the previous session to be persistent we can have a separate localStorage item `prevLog` and shift the logs accordingly. ie.
```
constructor(electronService) {
  this.logger = electronService.isElectron() ? window.require('electron-log') : console;
  const oldLog = localStorage.getItem(this.LOG_KEY);
  if (oldLog) {
    localStorage.setItem("prevLog", oldLog);
  }
  localStorage.setItem(this.LOG_KEY, "");
}

saveToLocalStorage(...params) {
  const existingLog = localStorage.getItem(this.LOG_KEY);
  localStorage.setItem(this.LOG_KEY, existingLog.concat(params));
}
```
So then we don't need to rely on the special limiter "AppConfig"

Although if there are better ideas feel free to go ahead!

Reply:
Mmm yes this can work but the purpose of the AppConfig is to trim the amount of data in the `prevLog` in your case. Might not be super applicable for the user's because at the moment i doubt they will reach the size threshold for individual `localStorage` data but for devs our `prevLog` will continuously grow in size unless we trim it.

I think maybe what we can do is have the first item on the log a custom text triggered by the LoggingService like idk maybe "Initializing CATcher" or something and we use that as a fixed item instead of relying `AppConfig` but yeah. Either way some soft of "trimming" mechanism is needed to prevent the amount of text from overshooting the threshold.

Reply:
Removed the use of regex and replaced it with a custom string that will signify the start of logs. Shifted the checks to constructor as well as opposed to everytime data is cached 👍